examples/gno.land/r/gnoland/blog/admin.gno

```diff
@@ -41,14 +41,46 @@ func AdminRemoveModerator(addr std.Address) {
 	moderatorList.Set(addr.String(), false) // FIXME: delete instead?
 }
 
-func NewPostExecutor(slug, title, body, publicationDate, authors, tags string) dao.Executor {
-	callback := func() error {
-		addPost(std.PreviousRealm().Address(), slug, title, body, publicationDate, authors, tags)
+type NewPostMetadata struct {
+	Slug            string
+	Title           string
+	Body            string
+	PublicationDate string
+	Authors         string
+	Tags            string
+	Caller          std.Address
+}
+
+func (a *NewPostMetadata) IsMetadata() {}
+func (a *NewPostMetadata) String() string {
+	return ufmt.Sprintf("- Post Title: %v\n- Post Publication Date: %v\n- Authors: %v\n- Tags: %v", a.Title, a.PublicationDate, a.Authors, a.Tags)
+}
 
-		return nil
+func newPostCallback(m dao.Metadata) error {
+	amm, ok := m.(*NewPostMetadata)
+	if !ok {
+		panic("wrong metadata type")
 	}
 
-	return bridge.GovDAO().NewGovDAOExecutor(callback)
+	addPost(amm.Caller, amm.Slug, amm.Title, amm.Body, amm.PublicationDate, amm.Authors, amm.Tags)
+
+	return nil
+}
+
+func NewPostProposalRequest(slug, title, body, publicationDate, authors, tags string) dao.ProposalRequest {
```

Comment:
what prevents me of doing:

```go
import "r/gnoland/blog"

prop := blog.NewPostPorposalRequest("slug", "title", "body", "date", "authors", "tags")
prop.Callback(prop.Metadata())
```

?

Reply:
@ajnavarro reopened. can you double-check please?

Reply:
NewPostPorposalRequest returns a ProposalRequest, that does not have a Build() method. ProposalRequestBuilder is used to generate a ProposalRequest. After that object is being generated, the content cannot be modified.

Reply:
https://github.com/gnolang/gno/pull/3389/files#diff-a67224ad1dfa87d6f696a2cf2f13e2db79fb6d3660d55af723aadfb012259f49R83-R89

Reply:
fixed 👍